recipes/kmod/all/conanfile.py

```diff
@@ -0,0 +1,105 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps, PkgConfigDeps
+import os
+
+required_conan_version = ">=1.44"
+
+class KModConan(ConanFile):
+    name = "kmod"
+    description = "linux kernel module handling library"
+    topics = ("kmod", "libkmod", "linux", "kernel", "module")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/kmod-project/kmod"
+    license = "LGPL-2.1-only"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+        "with_zstd": [True, False],
+        "with_xz": [True, False],
+        "with_zlib": [True, False],
+        "with_openssl": [True, False],
+        "experimental": [True, False],
+        "logging": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+        "with_zstd": True,
+        "with_xz": True,
+        "with_zlib": True,
+        "with_openssl": True,
+        "experimental": False,
+        "logging": False,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def generate(self):
+        yes_no = lambda v: "yes" if v else "no"
+        tc = AutotoolsToolchain(self)
+        tc.default_configure_install_args = True
+        tc.configure_args.append("--with-zstd=%s" % yes_no(self.options.with_zstd))
+        tc.configure_args.append("--with-xz=%s" % yes_no(self.options.with_xz))
+        tc.configure_args.append("--with-zlib=%s" % yes_no(self.options.with_zlib))
+        tc.configure_args.append("--with-openssl=%s" % yes_no(self.options.with_openssl))
+        tc.configure_args.append("--enable-experimental=%s" % yes_no(self.options.experimental))
+        tc.configure_args.append("--enable-logging=%s" % yes_no(self.options.logging))
+        tc.configure_args.append("--enable-debug=%s" % yes_no(self.settings.build_type == "Debug"))
+        tc.configure_args.append("--enable-tools=no")
+        tc.configure_args.append("--enable-manpages=no")
+        tc.configure_args.append("--enable-test-modules=no")
+        tc.configure_args.append("--enable-python=no")
+        tc.configure_args.append("--enable-coverage=no")
+        tc.generate()
+        tc = PkgConfigDeps(self)
+        tc.generate()
+        tc = AutotoolsDeps(self)
+        tc.environment.define("PKG_CONFIG_PATH", self.source_folder)  # is it really needed?
+        tc.environment.define("LIBS", "-lpthread")
+        tc.environment.define("libzstd_LIBS", "-lzstd")
```

Comment:
some workarounds for https://github.com/conan-io/conan/issues/10640 & https://github.com/conan-io/conan/issues/10341
in 1.44, `libzstd.pc` recursively references itself:
```
at /home/conan/.conan/data/kmod/29/_/_/build/e7842461bbcd29441fdb340a5299c019eae366ea/libzstd.pc
Name: libzstd
Description: Conan package: libzstd
Version: 1.5.2
Requires: libzstd
```

Reply:
okay, I have added, just in case